src/Sentry.Unity.Editor/AutoInstrumentation/ModuleReadWriter.cs

```diff
@@ -1,39 +1,125 @@
 using System;
-using System.Reflection;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
 using Mono.Cecil;
-using Mono.Cecil.Cil;
-using UnityEngine;
 
 namespace Sentry.Unity.Editor
 {
-    static class ModuleReaderWriter
+    internal class SentryPlayerReaderWriter
     {
-        public static (ModuleDefinition module, bool hasSymbols) Read(string file)
+        private const string PlayerAssemblyName = "Assembly-CSharp.dll";
+        private const string SentryUnityAssemblyName = "Sentry.Unity.dll";
+
+        private readonly string _playerAssemblyPath;
+        private readonly string _sentryUnityAssemblyPath;
+
+        private bool _playerModuleHasSymbols;
+        private ModuleDefinition _playerModule = null!;         // Set when reading the assemblies
+        private ModuleDefinition _sentryUnityModule = null!;    // Set when reading the assemblies
+
+        private readonly SentryAssemblyResolver _assemblyResolver;
+
+        internal SentryPlayerReaderWriter(
```

Comment:
This constructor and several methods are marked `internal` but seem like they could be `private` - or were you planning to use them elsewhere?